src/ride/track.c

```diff
@@ -4921,15 +4922,493 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+
+		uint32 temp_edx = RCT2_GLOBAL(0x993cfc, uint8*)[edx];
+		if ((temp_edx & 0xFF) == 0xFF) {
+			return loc_6CDDA2(x, y, rideIndex, mapElement);
+		}
+
+		uint16 xRounded = floor2(x, 32);
+		uint16 yRounded = floor2(y, 32);
+
+		uint32 ebx = direction;
+		xRounded -= TileDirectionDelta[direction].x;
+		yRounded -= TileDirectionDelta[direction].y;
+
+		uint8 bh = rideIndex;
+		bl = (uint8) ((z >> 3) & 0xFF);
+
+		rct_map_element *temp_esi = map_get_first_element_at(xRounded / 32, yRounded / 32);
+
+		// 0x6cdd71:
+		do {
+			if (map_element_get_type(temp_esi) != MAP_ELEMENT_TYPE_TRACK) {
+				continue;
+			}
+
+			if (temp_esi->base_height != bl) {
+				continue;
+			}
+
+			if (temp_esi->properties.track.ride_index != rideIndex) {
+				continue;
+			}
+
+			if (temp_esi->properties.track.type != 0x65) {
+				continue;
+			}
+
+			// loc_6CDD9B;
+			temp_esi->properties.track.maze_entry &= ~(1 << edx);
+
+			return loc_6CDDA2(x, y, rideIndex, mapElement);
+		} while (map_element_is_last_for_tile(temp_esi++));
+
+		// 6cdd92
+		mapElement->properties.track.maze_entry &= (1 << edx);
+		return loc_6CDDA2(x, y, rideIndex, mapElement);
+	}
+
+
+	// 0x6cdb53
+	if (direction == 4) {
+		return loc_6CDDA2(x, y, rideIndex, mapElement);
+	}
+
+	// 0x6cdb69
+
+	uint16 ax = x - TileDirectionDelta[direction].x / 2;
+	uint16 cx = x - TileDirectionDelta[direction].y / 2;
+
+	bl = (((mode << 8) + rideIndex) >> 3) & 0xFF;
+	// esi != mapElement
+	uint16 si = cx;
+
+	si = floor2(si, 32);
+	uint16 tmp_ax = floor2(ax, 32);
+
+	uint32 esi = ((si << 8) | tmp_ax) >> 3;
+
+	mapElement = gMapElementTilePointers[esi];
+
+	// eax = incremented X;
+	map_invalidate_tile_full(floor2(ax, 32), floor2(cx, 32));
+
+	// eax = ax;
+	// ecx = cx;
+	// ebx = [bl: bl, bh: rideIndex];
+
+	do {
+		if (map_element_get_type(mapElement) != MAP_ELEMENT_TYPE_TRACK) {
+			continue;
+		}
+
+		if (mapElement->base_height != bl) {
+			continue;
+		}
+
+		if (mapElement->properties.track.ride_index != rideIndex) {
+			continue;
+		}
+
+		if (mapElement->properties.track.type != 0x65) {
+			continue;
+		}
+
+		return loc_6CDBE9(x, y, ax, cx, z, rideIndex, mapElement);
+		// loc_6CDBDB:
+	} while (map_element_is_last_for_tile(mapElement++));
+
+	return loc_6CDDA2(x, y, rideIndex, mapElement);
+}
+
+
+money32 do_game_command_set_maze_track(uint16 x, uint8 flags, uint8 direction, uint16 y, uint8 rideIndex, uint8 mode, uint16 unknown, uint16 z) {
+	uint16 temp_dx;
+
+	// 0x006CD8CE: game_command_set_maze_track
+	RCT2_GLOBAL(RCT2_ADDRESS_NEXT_EXPENDITURE_TYPE, uint8) = 0;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_X, uint16) = x + 8;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_Y, uint16) = y + 8;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_Z, uint16) = z;
+
+	RCT2_GLOBAL(0xF4413E, money32) = 0;
+
+	if (!sub_68B044()) {
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if (!(flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) && RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0) {
```

Comment:
try to be consistent in your use of `flags` you seem to swap between `!(flags & GAME_COMMAND_FLAG_...)` and `(flags & GAME_COMMAND_FLAG_...) == 0)`.

I personally prefer:
`(flags & GAME_COMMAND_FLAG_...)` for if its set and
`!(flags & GAME_COMMAND_FLAG_...)` for if its not set.